sdk/cosmos/azure-cosmos/docs/StatusCodes.md

```diff
@@ -0,0 +1,53 @@
+# Cosmos DB Java SDK - Status Codes
+
+
+
+## Overview
+
+Below is a summary of the different status code / sub status code combinations that developers can experience in their application or diagnostics when using the Cosmos DB Java SDK. The column "Always transient" indicates whether this error condition is to be expected to always be transient (and auto-mitigated within seconds or at least few minutes). For error conditions that are not always transient the description contains additional context on what to watch out for if this error is seen for longer period of times. For errors that are supposed to be "always transient" it will be necessary to file a support ticket if the error condition exists for an extended period of time. The "Additional Info" column will also point to known issues if available and possible mitigations.
+
+
+
+## Status codes
+
+| Status code | Substatus code | Expecte to be transient | Additional info |
+| -----------------:|-----------------------:|:-----------------------:|:----------------------|
+|200|0|No| `OK`|
+|201|0|No| `Created` - returned for createItem or upsertItem when a new document was created|
+|204|0|No| `No Content` - returned when no payload would ever be returned - like for delete operations|
+|304|0|No| `Not Modified` - will be returned for `ReadFeed` and `ChangeFeed` operations to indicate that there are no more changes|
+|400|\*|No| `Bad Request` - indicates that the client violated some protocol constraint. See [Bad Request TSG - trouble-shooting guide](https://learn.microsoft.com/azure/cosmos-db/nosql/troubleshoot-bad-request) for more details.|
+|400|1001|No| `Bad Request/Partition key mismatch` - indicates that the PartitionKey defined in the point operation does not match the partition key value being extracted in the service form the document's payload based on the `PartitionKeyDefinition` of the container. See [Bad Request TSG - trouble-shooting guide](https://learn.microsoft.com/azure/cosmos-db/nosql/troubleshoot-bad-request) for more details.|
```

Comment:
400/1004: CrossPartitionQueryNotServable from gateway (esp with older SDK's)

Reply:
Thanks - added in next iteration